GDAXSharp/GDAXClient.cs

```diff
@@ -15,25 +15,29 @@
 using GDAXSharp.Services.Withdrawals;
 using GDAXSharp.Shared.Utilities.Clock;
 using GDAXSharp.Shared.Utilities.Queries;
+using System.Threading;
 
 namespace GDAXSharp
 {
     public class GDAXClient
     {
         public GDAXClient(
             IAuthenticator authenticator,
-            bool sandBox = false) 
-                    : this(authenticator, new HttpClient(), sandBox)
+            CancellationToken cancellationToken,
```

Comment:
so this is a breaking change. do we want them to always have to pass in a `CancellationToken`?

Reply:
I think we do, the purpose of the `cancellationToken` is to stop the `WebSocket`. Perhaps this is not the appropriate way, perhaps a boolean parameter is better way?

Reply:
Maybe don't pass the cancellationToken in via the WebSocket constructor.... meaning you don't need it passing into the GDAXClient.

Pass the cancellationToken in here:
 public void Start(List<ProductType> productTypes, List<ChannelType> channelTypes = null)

Or hold the cancellationToken internally within WebSocket and provide a public void Stop() to do the cancelling ?

Reply:
I think the latter option seems the most appropriate one 😄